python/core/qgsfields.sip

```diff
@@ -1,253 +1,333 @@
+ Alias for indexOf
+
+ \param fieldName The name of the field.
+
+ :return: The field index if found or -1 in case it cannot be found.
+.. seealso:: lookupField For a more tolerant alternative.
+ :rtype: int
+%End
+
+    int indexOf( const QString &fieldName ) const;
+%Docstring
+ Get the field index from the field name.
+ This method takes is case sensitive and only matches the data source
+ name of the field.
+
+ \param fieldName The name of the field.
+
+ :return: The field index if found or -1 in case it cannot be found.
+.. seealso:: lookupField For a more tolerant alternative.
+.. versionadded:: 3.0
+ :rtype: int
+%End
+
+    int lookupField( const QString &fieldName ) const;
+%Docstring
+ Look up field's index from the field name.
+ This method matches in the following order:
+
+  1. The exact field name taking case sensitivity into account
+  2. Looks for the field name by case insensitive comparison
+  3. The field alias (case insensitive)
+
+ \param fieldName The name to look for.
+
+ :return: The field index if found or -1 in case it cannot be found.
+.. seealso:: indexFromName For a more performant and precise but less tolerant alternative.
+.. versionadded:: 2.4
+ :rtype: int
+%End
+
     QgsAttributeList allAttributesList() const;
+%Docstring
+.. versionadded:: 2.4
+ :rtype: QgsAttributeList
+%End
 
-    //! Utility function to return a list of QgsField instances
     QList<QgsField> toList() const;
+%Docstring
+Utility function to return a list of QgsField instances
+ :rtype: list of QgsField
+%End
 
-    //! @note added in 2.6
-    bool operator==( const QgsFields& other ) const;
-    //! @note added in 2.6
-    bool operator!=( const QgsFields& other ) const;
+    bool operator==( const QgsFields &other ) const;
+%Docstring
+.. versionadded:: 2.6
+ :rtype: bool
+%End
+    bool operator!=( const QgsFields &other ) const;
+%Docstring
+.. versionadded:: 2.6
+ :rtype: bool
+%End
 
-    /** Returns an icon corresponding to a field index, based on the field's type and source
-     * @note added in QGIS 2.14
-     */
     QIcon iconForField( int fieldIdx ) const /Factory/;
-    %MethodCode
-      if ( a0 < 0 || a0 >= sipCpp->count() )
-      {
-        PyErr_SetString(PyExc_KeyError, QByteArray::number(a0));
-        sipIsErr = 1;
-      }
-      else
-      {
-        sipRes = new QIcon( sipCpp->iconForField( a0 ) );
-      }
-    %End
-
-    //! Allows direct construction of QVariants from fields.
+%Docstring
+ Returns an icon corresponding to a field index, based on the field's type and source
+.. versionadded:: 2.14
+ :rtype: QIcon
+%End
+%MethodCode
+    if ( a0 < 0 || a0 >= sipCpp->count() )
+    {
+      PyErr_SetString( PyExc_KeyError, QByteArray::number( a0 ) );
+      sipIsErr = 1;
+    }
+    else
+    {
+      sipRes = new QIcon( sipCpp->iconForField( a0 ) );
+    }
+%End
+
     operator QVariant() const;
+%Docstring
+Allows direct construction of QVariants from fields.
+%End
 
-/*  SIP_PYOBJECT __getitem__(int key);
```

Comment:
This was in comment and was falling back in a dockstring. This should be handled by operator[].